### PR TITLE
Clean up grakn.properties handling

### PR DESCRIPTION
**Pull request by aelred**
This involves making a class `GraknConfigKey` to represents key in `grakn.properties`. We move all parsing behaviour to the key, such as the type of the key (`Integer`, `String`, `Path`) and whether there are default values.

I've moved a lot of default values into the properties file itself. This way we don't have hard-coded default values in the code.

### CODE REVIEW

File: grakn-core/src/main/java/ai/grakn/GraknConfigKey.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import java.util.stream.Collectors;
+
+/**
+ * Class for keys of properties in the file {@code grakn.properties}.
+ *
+ * @param <T> the type of the values of the key
+ * @author Felix Chapman
+ */
+@AutoValue
+public abstract class GraknConfigKey<T> {
```
Comment (fppt) [inline grakn-core/src/main/java/ai/grakn/GraknConfigKey.java]: I am not going mad am I? Using Autovalue you are somehow initialising an abstract class??

Comment (aelred) [reply grakn-core/src/main/java/ai/grakn/GraknConfigKey.java]: That's how autovalue works 👍 
It generates a concrete implementation of the class, called `AutoValue_GraknConfigKey`

File: conf/test/tinker/grakn.properties [hunk 2]
```diff
@@ -58,3 +61,5 @@ loader.threads=1
 password.protected=false
 JWT.secret=Anzdz9l4TG8N5y18oaTDueqJQca1aX7loaz0U8Hz
 admin.password=grakn
+
+test.start.embedded.components=true
```
[the remark this answers is not in the document]
Comment (aelred) [reply conf/test/tinker/grakn.properties]: I just looked at the code and it seems when it's false, the `EngineContext` won't start engine or Redis. I suppose that's for if you want to run the tests against a running instance?

Nothing in the code ever sets it to false, so I don't know if it's worth keeping.

[the remark this answers is not in the document]
Comment (aelred) [reply conf/test/tinker/grakn.properties]: Oh wait! I take it back - it's false in the cluster config. I don't know what that means though.

**Comment by aelred** [comment]
!rtg - some error in `grakn-dist`

**Comment by aelred** [comment]
!rtg - surprising error

**Pull request merged**